src/pgen/cluster.cpp

```diff
@@ -405,4 +432,101 @@ void ProblemGenerator(MeshBlock *pmb, parthenon::ParameterInput *pin) {
   } // END if(hydro_pkg->Param<Fluid>("fluid") == Fluid::glmmhd)
 }
 
+void UserWorkBeforeOutput(MeshBlock *pmb, ParameterInput *pin) {
+  // get hydro
+  auto pkg = pmb->packages.Get("Hydro");
+  const Real gam = pin->GetReal("hydro", "gamma");
+  const Real gm1 = (gam - 1.0);
+
+  // get prim vars
+  auto &data = pmb->meshblock_data.Get();
+  auto const &prim = data->Get("prim").data;
+
+  // get derived fields
+  auto &log10_radius = data->Get("log10_cell_radius").data;
+  auto &entropy = data->Get("entropy").data;
+  auto &mach_sonic = data->Get("Mach_sonic").data;
+  auto &temperature = data->Get("temperature").data;
+
+  // for computing temperature from primitives
+  auto mbar_over_kb = pkg->Param<Real>("mbar_over_kb");
+
+  // fill derived vars (*including ghost cells*)
+  auto &coords = pmb->coords;
+  IndexRange ib = pmb->cellbounds.GetBoundsI(IndexDomain::entire);
+  IndexRange jb = pmb->cellbounds.GetBoundsJ(IndexDomain::entire);
+  IndexRange kb = pmb->cellbounds.GetBoundsK(IndexDomain::entire);
+
+  pmb->par_for(
+      "FillDerivedHydro", kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int k, const int j, const int i) {
+        // get gas properties
+        const Real rho = prim(IDN, k, j, i);
+        const Real v1 = prim(IV1, k, j, i);
+        const Real v2 = prim(IV1, k, j, i);
+        const Real v3 = prim(IV1, k, j, i);
+        const Real P = prim(IPR, k, j, i);
+
+        // compute radius
+        const Real x = coords.Xc<1>(i);
+        const Real y = coords.Xc<2>(j);
+        const Real z = coords.Xc<3>(k);
+        const Real r2 = SQR(x) + SQR(y) + SQR(z);
+        log10_radius(0, k, j, i) = 0.5*std::log10(r2);
```

Comment:
Is the leading 0 index required? (also for the other vars)?

Reply:
I don't know - are the array accessors designed to work with just `(k,j,i)`?

Reply:
Yes, they should. Internally it's a 7D array that you could reference with up to 7 indexers.